fea-rs/src/compile/lookups/gsub_builders.rs

```diff
@@ -67,82 +54,24 @@ impl Builder for SingleSubBuilder {
     type Output = Vec<write_gsub::SingleSubst>;
 
     fn build(self, _: &mut VariationStoreBuilder) -> Self::Output {
-        const COST_OF_EXTRA_SUB1F1_SUBTABLE: usize = 2 + // extra offset
-            2 + 2 + 2 + // format1 table itself
-            2 + 2; // extra coverage table
-
-        const N_GLYPHS_TO_JUSTIFY_EXTRA_SUB1F1: usize =
-            COST_OF_EXTRA_SUB1F1_SUBTABLE / GlyphId16::RAW_BYTE_LEN;
-
-        #[derive(Default)]
-        struct SubtableMap {
-            // key is the delta between the two gylphs.
-            format1: BTreeMap<i16, Vec<(GlyphId16, GlyphId16)>>,
-            format2: Vec<(GlyphId16, GlyphId16)>,
+        if self.items.is_empty() {
+            return Default::default();
         }
+        let delta = self
+            .items
+            .iter()
+            .map(|(k, v)| v.to_u16() as i32 - k.to_u16() as i32)
+            .reduce(|acc, val| if acc == val { acc } else { i32::MAX })
```

Comment:
maybe I'm just slow today but this reduce is throwing me, please add a comment explaining what's happening here

Reply:
...I just realized the comment in question is below (re equidistant), maybe move it up slightly?